### PR TITLE
Feat loss 11

### PR DESCRIPTION
**Pull request by mingzailao**
For str, i think it should be implemented at last.
for loss, i modify the loss to compute the loss for each instance and return the list of each instance's loss. is it OK? or mean loss is better?

### CODE REVIEW

File: tfs/core/regularizers/__init__.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+                res=tf.add(res,tf.reduce_sum(tmp,keep_dims=False))
+        return res
+
+    def __str__(self):
+        return ""
+
+def DefaultRegularizer(netobj):
+    return L1(netobj,l1=0.01,nodes_params=None)
+
+
```
Comment (crackhopper) [inline tfs/core/regularizers/__init__.py]: please set default regularizer to be none (with no calculation).  

and add test about none regularizer.  (it should be working consistently with the default loss function)

File: tfs/network/base.py [hunk 2]
```diff
@@ -244,13 +258,44 @@ def _build(self,input_shape,dtype,idx=None):
...
+    return self._loss
+
+  @property
+  def regulization(self):
+    if self._regulization is None:
+      self._compute_regulization()
+    return self._regulization
+
+  @property
+  def objective(self):
```
Comment (crackhopper) [inline tfs/network/base.py]: no need of objective.  The training objective is just to minimize the loss function.  so calculate loss by adding the regularizer output.

File: tests/core/loss/loss_test.py [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+    def test_categoricalcrossentropy(self,capsys):
+        netobj=LeNet()
+        netobj.build([100,28,28,1])
+        assert netobj.true_output.get_shape().as_list()==netobj.output.get_shape().as_list()
+        assert netobj.loss.get_shape().as_list()==[100]
+        netobj=TenoutNet()
+        netobj.build([100,28,28,1])
+        assert netobj.true_output.get_shape().as_list()==netobj.output.get_shape().as_list()
+        assert netobj.loss.get_shape().as_list()==[100]
+
```
Comment (crackhopper) [inline tests/core/loss/loss_test.py]: please add the codes to eval the loss.  (we should make sure the loss can be evaluated by feed input and true output)

File: tests/core/regularizers/test_regularizers.py [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+        netobj.build()
+        assert netobj.regulization.graph==netobj.graph
+
+    def test_l1(self,capsys):
+        netobj=LeNet(in_shape=[1,28,28,1])
+        netobj.Regularizer=L1(netobj,l1=0.01,nodes_params=[0,0,0,0.01])
+        netobj.build()
+        assert netobj.Regularizer.nodes_params==[0,0,0,0.01]
+        assert netobj.regulization.graph==netobj.graph
+
```
Comment (crackhopper) [inline tests/core/regularizers/test_regularizers.py]: better add evaluation of the regularizer output.

**Comment by crackhopper** [comment]
"For str, i think it should be implemented at last."
- it's not that important.  But it should have.

"for loss, i modify the loss to compute the loss for each instance and return the list of each instance's loss. is it OK? or mean loss is better?"
- a list of loss is better. Because we also set back propagation also on each device.  the mean loss is useless, because we back propagation from each loss on each device to calculate the gradient, and the average the gradient to calculate the parameter modification.

File: tests/core/loss/loss_test.py [hunk 5]
```diff
@@ -4,6 +4,13 @@
 from tfs.models import LeNet
 from tfs.core.loss import Loss,CategoricalCrossentropy,DefaultLoss,BinaryCrossentripy,SquareError
 from tfs.network.base import CustomNetwork
+from tensorflow.examples.tutorials.mnist import input_data
+from tfs.core.regularizers import Regularizer,L1,L2
+
+mnist=input_data.read_data_sets("~/.keras/datasets",one_hot=True)
```
Comment (crackhopper) [inline tests/core/loss/loss_test.py]: then the test can only pass on your computer.  

I will fix this one. 

remember to avoid using outside dataset without providing automatic downloading script.
